propagator/ottrace/lib/opentelemetry/propagator/ottrace/text_map_extractor.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+module OpenTelemetry
+  module Propagator
+    module OTTrace
+      class TextMapExtractor
+        PADDING = '0' * 16
+        VALID_TRACE_ID_REGEX = /^[0-9a-f]{32}$/.freeze
+        VALID_SPAN_ID_REGEX = /^[0-9a-f]{16}$/.freeze
+
+        # Returns a new TextMapExtractor that extracts OTTrace context using the
+        # specified getter
+        #
+        # @param [optional Getter] default_getter The default getter used to read
+        #   headers from a carrier during extract. Defaults to a
+        #   {OpenTelemetry::Context:Propagation::TextMapGetter} instance.
+        # @return [TextMapExtractor]
+        def initialize(
+          baggage_manager:,
+          default_getter: Context::Propagation.text_map_getter
+        )
+          @baggage_manager = baggage_manager
+          @default_getter = default_getter
+        end
+
+        # Extract OTTrace context from the supplied carrier and set the active span
+        # in the given context. The original context will be returned if OTTrace
+        # cannot be extracted from the carrier.
+        #
+        # @param [Carrier] carrier The carrier to get the header from.
+        # @param [Context] context The context to be updated with extracted context
+        # @param [optional Getter] getter If the optional getter is provided, it
+        #   will be used to read the header from the carrier, otherwise the default
+        #   getter will be used.
+        # @return [Context] Updated context with active span derived from the header, or the original
+        #   context if parsing fails.
+        def extract(carrier, context, getter = nil)
+          getter ||= default_getter
+          trace_id = getter.get(carrier, TRACE_ID_HEADER)
+          span_id = getter.get(carrier, SPAN_ID_HEADER)
+          sampled = getter.get(carrier, SAMPLED_HEADER)
+
+          return context unless trace_id && span_id
+
+          trace_id = trace_id.length == 16 ? "#{PADDING}#{trace_id}" : trace_id
+
+          return context if VALID_TRACE_ID_REGEX !~ trace_id || VALID_SPAN_ID_REGEX !~ span_id
```

Comment:
i think this is fine but i'm curious what the use case is for the `else` case here? Can OT Tracing headers be either 64 or 128 bit?

Reply:
I do not know. This is a bit of copy+paste action taken from the JS and Golang Implementations:

https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/propagators/opentelemetry-propagator-ot-trace/src/OTTracePropagator.ts#L91

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/propagators/ot/ot_propagator.go#L137

Reply:
I wrote one of the originals. Now that you point it out, I don't think the else case is possible under normal circumstances. We could leave it here out of fear or remove it.

Reply:
If it is ok with you both @mwear and @ericmustin I am going to leave it here so it is symmetric with the other implementations.

Reply:
👍